### PR TITLE
First pass at empty state page

### LINKED ISSUE #3278: Empty State - Add page to Design System site with guidance and examples
### What is the improvement or addition you'd like to see?

Follow on to: https://github.com/grommet/hpe-design-system/issues/3221

Deliverables:
- [ ] Add page to Design System site with guidance and examples based on https://www.figma.com/file/4uEMfLl9riZDJ4y6dlv1Jq/HPE-Empty-State?node-id=2729-13258&t=7Y8ZaKmKM3FMfEYb-4

### Can you describe any use cases driving your request?

_No response_

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-3281--keen-mayer-a86c8b.netlify.app/templates/empty-state)

#### What does this PR do?

Implementation of [empty state guidance](https://www.figma.com/file/4uEMfLl9riZDJ4y6dlv1Jq/HPE-Empty-State?node-id=2729-13258&t=HLDOTNsQS5Hjcrc1-0).

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
Closes #3278

### CODE REVIEW

File: aries-site/src/pages/templates/empty-state.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  <EmptyStateExample />
+</Example>
+
+## When to use
+
+Common empty state use cases include:
+
+- “Day 0” use cases when a user first starts using an application
+- When the user completes an action that returns no results
+- When an error prevents the system from returning any data
```
Comment (KennyAtHPE) [inline aries-site/src/pages/templates/empty-state.mdx:24]: System-generated, task-generated, or both kinds of error? I'm a little confused on when to use notifications and when to use empty states here or whether or not there are opportunities to use both.

File: aries-site/src/pages/templates/empty-state.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+It is our responsibility as designers to provide our users with clear, concise guidance on why the data isn’t present and how to move forward in their journey.
+
+A well-designed empty state can improve the user experience and reduce confusion by setting clear expectations for what the user should do next.
+
+Here are a few characteristics that make a well-designed empty state:
+
+1.  **Clear messaging:** The empty state should have simple and clear messaging that tells the user what information is missing, why it is missing, and what they can do next to access the information. Follow the voice and tone guidance within the design system to remain clear and empathetic to the user.
```
Comment (KennyAtHPE) [inline aries-site/src/pages/templates/empty-state.mdx:32]: Directly link to voice and tone page?

File: aries-site/src/pages/templates/empty-state.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+1.  **Visual cues:** The empty state should use iconography to help convey the message and create a more engaging user experience. The icon should be relevant to the context of the interface and help the user understand why it's empty.
```
Comment (KennyAtHPE) [inline aries-site/src/pages/templates/empty-state.mdx:33]: Wondering if we should just provide a handful of icons that folks should pick from. Realistically, there are only a few I could think of that would be used for empty states.

File: aries-site/src/pages/templates/empty-state.mdx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+| **1** | **Icon**        | Imagery to help the user process the message.                       |                                             | Use size="xlarge" on icon.                                                 |
+| **2** | **Heading**     | Communicates the system status.                                     | <Checkmark a11yTitle="true" size="small" /> | Use the semantic heading level that fits the placement of the empty state. |
+| **3** | **Description** | Guides the user out of the empty state                              | <Checkmark a11yTitle="true" size="small" /> | Use default text size of "medium".                                         |
+| **4** | **Action**      | A direct pathway out of the empty state.                            |                                             |                                                                            |
+| **5** | **Anchor**      | An additional learning cue to help the user out of the empty state. |                                             |                                                                            |
```
Comment (KennyAtHPE) [inline aries-site/src/pages/templates/empty-state.mdx:52]: I know we've been leaning towards using the term 'Link' instead of 'Anchor' ... should that be changed here as well?

File: aries-site/src/pages/templates/empty-state.mdx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+| ----- | --------------- | ------------------------------------------------------------------- | :-----------------------------------------: | -------------------------------------------------------------------------- |
+| **1** | **Icon**        | Imagery to help the user process the message.                       |                                             | Use size="xlarge" on icon.                                                 |
+| **2** | **Heading**     | Communicates the system status.                                     | <Checkmark a11yTitle="true" size="small" /> | Use the semantic heading level that fits the placement of the empty state. |
+| **3** | **Description** | Guides the user out of the empty state                              | <Checkmark a11yTitle="true" size="small" /> | Use default text size of "medium".                                         |
+| **4** | **Action**      | A direct pathway out of the empty state.                            |                                             |                                                                            |
```
Comment (KennyAtHPE) [inline aries-site/src/pages/templates/empty-state.mdx:51]: For Action and Anchor, I would probably state within the notes to try and keep it to two or fewer.

**Review by KennyAtHPE — COMMENTED** [review]
Excellent job on this. The overall structure of the page flows nicely. I would have liked to see a do's and don't section specifically detailing:
1. Do/Don't on how to write effective yet short messages, how many links/actions to include, and what icon to use.  
2. How this could be used or not used with Notifications. The use cases "After completing a task" and "When an error prevents the system from returning any data" sound to me like they could also be displayed through toasts and inline notifications.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 4 above)
Comment (taysea) [reply aries-site/src/pages/templates/empty-state.mdx:52]: I might've missed that -- happy to adjust if that's the desired direction. Can you fill me in on when we shifted towarads that? The reason I used "Anchor" is because that aligns with the component name in Grommet and Figma.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 3 above)
Comment (taysea) [reply aries-site/src/pages/templates/empty-state.mdx:33]: If you have a shortlist, we can add. I think for the first pass of the page it's sufficient to say the icon should be relevant to the context. I'm not sure we want to be in the business of trying to cover every context that would have an empty state.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 1 above)
Comment (taysea) [reply aries-site/src/pages/templates/empty-state.mdx:24]: Can you draft what you might expect this to say?

**Comment by taysea** [comment]
> 2. How this could be used or not used with Notifications. The use cases "After completing a task" and "When an error prevents the system from returning any data" sound to me like they could also be displayed through toasts and inline notifications.

@KennyAtHPE I'd value your perspective/drafted guidance on how to differentiate when to use.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 2 above)
Comment (taysea) [reply aries-site/src/pages/templates/empty-state.mdx]: Done.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 5 above)
Comment (taysea) [reply aries-site/src/pages/templates/empty-state.mdx]: Done.

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export const EmptyState = ({
+  actions,
+  align,
+  description,
+  icon,
+  level,
+  title,
+}) => {
+  return (
```
Comment (halocline) [inline aries-core/src/js/components/layouts/EmptyState/EmptyState.js:13]: Would something like this be useful so that the EmptyState layout component has xlarge by default?

```suggestion
 if (!icon.size) {
   icon.size="xlarge"
 }

  return (
```

File: aries-site/src/examples/templates/empty-state/AccessDenied.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import { Lock } from 'grommet-icons';
+import { EmptyState } from 'aries-core';
+
+export const AccessDenied = () => {
+  return (
+    <EmptyState
+      title="Access denied"
+      description={`You do not have access to this resource. 
+    Please contact the administrator to request access.`}
+      icon={<Lock size="xlarge" />}
```
Comment (halocline) [inline aries-site/src/examples/templates/empty-state/AccessDenied.js:12]: See related comment in EmptyState.js -- just one less piece of written guidance for us/callers to have to remember.

File: aries-site/src/examples/templates/empty-state/AccessDenied.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+      actions={<Button label="Contact administator" primary />}
```
Comment (halocline) [inline aries-site/src/examples/templates/empty-state/AccessDenied.js:13]:
```suggestion
      actions={<Button label="Request access" primary />}
```

File: aries-site/src/examples/templates/empty-state/ActionEmptyState.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            <Button label="Return to dashboard" primary />
+            <Anchor label="When will I hear back?" />
+          </Box>
+        }
+        level={2}
+      />
+      <EmptyState
+        title="All notifications cleared"
+        description={`There are no further notifications 
+        for you to vie.`}
```
Comment (halocline) [inline aries-site/src/examples/templates/empty-state/ActionEmptyState.js:32]:
```suggestion
        for you to view.`}
```

File: aries-site/src/pages/templates/empty-state.mdx (hunk 3 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx:33]: Agree with avoiding trying to over-engineer. If we do have a short-list in mind, we could add a simple sentence saying "Some more common icons are ...." That said, I think cases where the icon may be related to the type of objects in a data collection (e.g. access points, servers, clusters, etc.) so leaving it as "relevant to the context" may be the best way to go for now.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 1 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx:24]: @KennyAtHPE , you bring up a good point about when do I use empty state vs. inline notifications?

I think there is an aspect to having:
1. no data, where the empty state is now taking the place of where content would normally be displayed. Whereas the inline notification is likely supplementing/augmenting content where some data exists.
1. Would generally be system-level messages, but a user action could also lead to an empty state (e.g. too strict of filter or search criteria).

File: aries-site/src/pages/templates/empty-state.mdx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+| Label | Region          | Purpose                                                             |                  Required                   | Notes                                                                                                                                   |
+| ----- | --------------- | ------------------------------------------------------------------- | :-----------------------------------------: | --------------------------------------------------------------------------------------------------------------------------------------- |
+| **1** | **Icon**        | Imagery to help the user process the message.                       |                                             | Use size="xlarge" on icon.                                                                                                              |
+| **2** | **Heading**     | Communicates the system status.                                     | <Checkmark a11yTitle="true" size="small" /> | Use the semantic heading level that fits the placement of the empty state.                                                              |
```
Comment (halocline) [inline aries-site/src/pages/templates/empty-state.mdx:49]:
```suggestion
| **2** | **Heading**     | Communicates the system status.                                     | <Checkmark a11yTitle="true" size="small" /> | <span>Use the [semantic heading level](/foundation/typography#semantic-usage-of-heading-levels) that fits the placement of the empty state.</span>                                                     |
```

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    <Box gap="medium" align={align || 'center'} flex={false}>
+      {icon}
+      <Box align={align || 'center'} gap="xsmall">
+        <Heading margin="none" level={level}>
+          {title}
+        </Heading>
+        <Paragraph margin="none" textAlign={align || 'center'}>
```
Comment (halocline) [inline aries-core/src/js/components/layouts/EmptyState/EmptyState.js:20]: 🤔  I'm wondering about the 'center' alignment of the textAlign...  I can't think of anywhere else where we do that. Should we always treat paragraphs as left justified?

@KennyAtHPE thoughts?

File: aries-site/src/pages/templates/empty-state.mdx [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+  <ActionEmptyState />
+</Example>
+
+### Error management empty state
+
+Error management empty states are displayed when there has been an error that prevents the content from loading or displaying correctly. These are designed to provide feedback to the user when an error occurs. This can include error messages or prompts that guide the user through the error and provide possible solutions or next steps.
+
+The goal of an error management empty state is to help the user understand what went wrong and how to resolve the issue. Some considerations for writing error management empty states:
+
+1. **Consistent visual cues:** these empty states typically use the “alert” status icon and “status. warning” color. Only deviate from this standard if it is necessary for your context.
```
Comment (halocline) [inline aries-site/src/pages/templates/empty-state.mdx:124]:
```suggestion
1. **Consistent visual cues:** these empty states typically use the “alert” status icon and “status warning” color. Only deviate from this standard if it is necessary for your context.
```

File: aries-site/src/pages/templates/empty-state.mdx (hunk 4 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx]: As well as semantic HTML.

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+export const EmptyState = ({
+  actions,
+  align,
+  description,
+  icon,
+  level,
+  title,
+}) => {
+  return (
+    <Box gap="medium" align={align || 'center'} flex={false}>
```
[the remark this answers is not in the document]
Comment (taysea) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js:14]: Good idea, I'll look at doing that

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js (hunk 6 above)
Comment (taysea) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js:13]: Yes, I think that's a good idea.

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js (hunk 11 above)
Comment (taysea) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js:20]: Based on share-out discussion, Chris seemed comfortable with center alignment for this use case.

Comment (halocline) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js:20]: Agree.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 1 above)
Comment (KennyAtHPE) [reply aries-site/src/pages/templates/empty-state.mdx:24]: @halocline I think generally those 2 points are a good distinction between inline notifications and empty states.

What I'm drawn most to within the current guidance is the use case "When an error prevents the system from returning any data." I feel as though I could see this exact bullet point under inline notifications. I'm concerned that designers would have an empty state and an inline notification that would both display the same message and any anchors/actions that would point to the same thing.

If these errors are user-generated, I would lean towards using inline notifications since it's a bit more intrusive and can better capture the user's attention. On the other hand, if it's a system-generated error, I could see an empty state being used instead. In both cases, I can see both being used together so it's a bit tricky depending on the use case.

Long story short though, for the time being, I would change that bullet point into something like this:
- When an error prevents the system from returning any data
      - For user-generated errors, [Inline Notifications](url) can be used together with empty states to provide additional feedback related to the user's actions.
      
I'd probably make a follow-up ticket to expand on this guidance and to provide visual examples as well.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 3 above)
Comment (KennyAtHPE) [reply aries-site/src/pages/templates/empty-state.mdx:33]: Looking at the use cases more, I agree with keeping them "relevant to the context." Originally thought it was mostly just status indicators, the error, close, and a few other misc. icons.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 4 above)
Comment (KennyAtHPE) [reply aries-site/src/pages/templates/empty-state.mdx]: @SOjaHPE any thoughts you can weigh in on here? I know we've discussed in the past how the term 'Link' might better fit a user's mental modal as opposed to 'Anchor'

File: aries-site/src/examples/templates/empty-state/ActionEmptyState.js (hunk 9 above)
Comment (taysea) [reply aries-site/src/examples/templates/empty-state/ActionEmptyState.js]: Done.

File: aries-site/src/examples/templates/empty-state/AccessDenied.js (hunk 8 above)
Comment (taysea) [reply aries-site/src/examples/templates/empty-state/AccessDenied.js:13]: Done.

File: aries-site/src/examples/templates/empty-state/AccessDenied.js (hunk 7 above)
Comment (taysea) [reply aries-site/src/examples/templates/empty-state/AccessDenied.js]: Done.

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js (hunk 6 above)
Comment (taysea) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js:19]: Done.

File: aries-core/src/js/components/layouts/EmptyState/EmptyState.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (taysea) [reply aries-core/src/js/components/layouts/EmptyState/EmptyState.js]: Done.

**Comment by KennyAtHPE** [comment]
> > 2. How this could be used or not used with Notifications. The use cases "After completing a task" and "When an error prevents the system from returning any data" sound to me like they could also be displayed through toasts and inline notifications.
> 
> @KennyAtHPE I'd value your perspective/drafted guidance on how to differentiate when to use.

Same comment in the above thread. I think (as a separate ticket to explore in the future) it'd be very helpful to include visual examples for both of these use cases as to how notifications can be used in conjunction with empty states. At the end of the day, we'd want these components to support each other rather than sharing out duplicative info. I don't have the exact guidance or to what the visuals would look like in my mind right now.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 4 above)
Comment (jcfilben) [reply aries-site/src/pages/templates/empty-state.mdx]: Some additional context from the grommet side... Eric and I have chatted about renaming the grommet component from 'Anchor' to 'Link'. This is the direction we would like to eventually head, but for backwards compatibility we couldn't make this change to the grommet component without doing a major version release. It is tracked in [this issue](https://github.com/grommet/grommet/issues/3297) as well as in our [grommet v3 wiki](https://github.com/grommet/grommet/wiki/Grommet-v3).
Totally fine if we want to refer to it as Link here, just wanted to give some background from the grommet side.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 5 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx]: Confirmed.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 4 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx]: Changed to Link.

File: aries-site/src/pages/templates/empty-state.mdx (hunk 2 above)
Comment (halocline) [reply aries-site/src/pages/templates/empty-state.mdx]: confirmed.

**Review by halocline — APPROVED** [review]
